core/vm/interpreter.go

```diff
@@ -198,6 +198,7 @@ func (in *EVMInterpreter) Run(contract *Contract, input []byte, readOnly bool) (
 	// they are returned to the pools
 	defer func() {
 		returnStack(stack)
+		mem.Free()
```

Comment:
Since we have a `func returnStack(s *Stack)` already, it would be nicer to make them aligned, and have `returnMemory(m *Memory)`.

Reply:
`NewMemory` is public and since it uses the pool it would be nice to also have the free method available. Should we maybe change the stack to Free as well or switch to `returnMemory` and make `NewMemory` private?